src/async_nbgrader/async_nbgrader/handlers.py

```diff
@@ -0,0 +1,88 @@
+import json
+import os
+import pika
+
+from nbgrader.server_extensions.formgrader.apihandlers import AutogradeHandler
+from nbgrader.server_extensions.formgrader.base import check_notebook_dir
+from nbgrader.server_extensions.formgrader.base import check_xsrf
+from notebook.notebookapp import NotebookApp
+from notebook.utils import url_path_join as ujoin
+from tornado import web
+
+
+class AsyncAutogradeHandler(AutogradeHandler):
+    @web.authenticated
+    @check_xsrf
+    @check_notebook_dir
+    def post(self, assignment_id: str, student_id: str) -> None:
+        """Handler for processing autograding request, queues autograding task in amqp"""
+        connection = pika.BlockingConnection(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *connection*. The resource is allocated by call *pika.BlockingConnection*. Execution paths that do not contain closure statements were detected. Close *connection* in a try-finally block to prevent the resource leak.